pipeline/measuring.py

```diff
@@ -414,3 +425,36 @@ def make_filter_bank(self, imsize, psf_fwhm):
         self._filter_bank = templates
         self._filter_psf_fwhm = psf_fwhm
 
+    def compare_measurement_to_thresholds(self, m):
+        """Compare measurement disqualifiers of a Measurements object to the thresholds set for 
+        this measurer object.
+
+        Inputs:
+          - m : a Measurements object to be compared
+
+        returns one of three strings to indicate the result
+          - "ok"     : All disqualifiers below both thresholds
+          - "bad"    : Some disqualifiers above mark_thresh but all 
+                       below deletion_thresh
+          - "delete" : Some disqualifiers above deletion_thresh
+        
+        """
+        passing_status = "ok"
+
+        mark_thresh = m.provenance.parameters["thresholds"] # thresholds above which measurement is marked 'bad'
+        deletion_thresh = self.pars.deletion_thresholds
+
+        # I think this accomplishes what we wanted without having to fully loop over both dictionaries
+        combined_keys = np.unique(list(mark_thresh.keys()) + list(deletion_thresh.keys())) # unique keys from both
+        for key in combined_keys:
+            # check if it exists in mark_ but not deletion_. If so, use the value from mark_ for deletion_
+            if deletion_thresh.get(key) is None and mark_thresh.get(key) is not None:
```

Comment:
Of note with this implementation: You can no longer set the entire `deletion_thresholds` parameter to `None` (it only accepts type `dict` so the equivalent is `{}`), and any missing deletion parameters will default to the value in the `thresholds` dict, regardless of whether they are missing or set to `None`. The only way to change that behavior e.g. if you wanted to turn deletion off would be to explicitly set a value for each disqualifier that would never be exceeded.

Reply:
i *think* this accomplishes what was discussed, with a focus on making it simple to make the deletion behavior just use the normal threshold when desired, and possible but not completely trivial to disable deletion if that is really desired.

Reply:
Updated as per Friday meeting discussion